tensorboard/webapp/runs/views/runs_table/runs_table_container.ts

```diff
@@ -353,9 +376,51 @@ export class RunsTableContainer implements OnInit {
       );
     }
 
+    /**
+     * For consumers who show checkboxes, notify users that new runs may not be
+     * selected by default. Avoid showing it more than once per route, since it
+     * would be annoying to see the alert on every auto-reload (assuming a new
+     * run per reload).
+     *
+     * Warning: this pattern is not recommended in general. Dispatching
+     * `alertReported` would be better handled in a Ngrx Reducer in response
+     * to `fetchRunsSucceeded` or via the declared alert registrations using
+     * `alertFromAction`. Unfortunately, those currently have no way of knowing
+     * whether a run table is actually shown (with checkboxes), so we make a
+     * special exception here. A more 'Ngrx pure' approach would require making
+     * the store aware of the visibility of any run tables.
+     */
+    if (this.columns.includes(RunsTableColumn.CHECKBOX)) {
+      const runsExceedLimitForRoute$ = this.store.select(getRouteId).pipe(
+        takeUntil(this.ngUnsubscribe),
+        switchMap(() => {
+          return rawAllUnsortedRunTableItems$.pipe(
+            filter((runTableItems: RunTableItem[]) => {
+              return runTableItems.length > MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT;
+            }),
+            take(1)
+          );
+        })
+      );
+      runsExceedLimitForRoute$.subscribe(() => {
+        const text =
+          `The number of runs exceeds ` +
+          `${MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT}. New runs are unselected ` +
+          `for performance reasons.`;
+        this.store.dispatch(
+          alertActions.alertReported({localizedMessage: text})
+        );
```

Comment:
Just wondering, does any asynchronusity disrupt the UX? Adding what is effectively `setTimeout(0)` would make me feel tiny bit better about the synchronous composite action.

Reply:
I manually added it to try it out, and the UX seems fine, e.g. Alert still appears and the rest of the UI is functional.  Not sure how valuable it is to go add setTimeouts to all cases like these, but perhaps worth considering in a separate PR.